pipelines/build/common/riscv_version_out.groovy

```diff
@@ -0,0 +1,116 @@
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
+/**
+ * This file is a jenkins job for extracting a version string from a cross-compiled, riscv binary.
+ * See https://github.com/AdoptOpenJDK/openjdk-build/issues/1773 for the inspiration for this.
+ *
+ * This file is referenced by the upstream job at pipelines/build/common/openjdk_build_pipeline.groovy.
+ * This job is run at build-scripts/job/utils/job/riscv-version-out/.
+ *
+ * The job looks like:
+ *  1. Switch into a riscv node
+ *  2. Retrieve the artifacts built in the upstream job
+ *  3. Run java -version and export to file
+ *  4. Expose to the upstream job by archiving file
+ */
+
+// TODO: ADD THE ACTIVE NODE TIMEOUT LOGIC HERE OR GET IT MERGED INTO JOB HELPER (https://github.com/AdoptOpenJDK/openjdk-build/issues/2235)
+
+node ("riscv&&ci.role.test") {
+    timestamps {
+        try {
+            Integer JOB_TIMEOUT = 1
```

Comment:
I suspect some of this can be extracted to local functions/methods or does the DSL prevent this?

Reply:
I stuck it in a separate file mostly to avoid clogging up `openjdk_build_pipeline.groovy`, as that one is quite big already. It doesn't have to be and can be moved across if people think it's better keeping it all together?